pkg/server/api/svid/v1/service.go

```diff
@@ -164,6 +159,28 @@ func (s *Service) BatchNewX509SVID(ctx context.Context, req *svid.BatchNewX509SV
 	return &svid.BatchNewX509SVIDResponse{Results: results}, nil
 }
 
+// fetchEntries fetches authorized entries using caller ID from context
+func (s *Service) fetchEntries(ctx context.Context, log logrus.FieldLogger) (map[string]*types.Entry, error) {
+	callerID, ok := rpccontext.CallerID(ctx)
+	if !ok {
+		log.Error("Caller ID missing from request context")
+		return nil, status.Error(codes.Internal, "caller ID missing from request context")
+	}
+
+	entries, err := s.ef.FetchAuthorizedEntries(ctx, callerID)
+	if err != nil {
+		log.WithError(err).Error("Failed to fetch registration entries")
+		return nil, status.Error(codes.Internal, "failed to fetch registration entries")
+	}
+
+	entriesMap := make(map[string]*types.Entry)
```

Comment:
super nit:Since the length is known, it may be worth pre-allocating map capacity to reduce allocation pressure? (the RPC that uses this is going to be 3rd on the most frequently list)
```suggestion
	entriesMap := make(map[string]*types.Entry, len(entries))
```